File: java/src/org/woahoverflow/chad/handle/audio/MusicHandler.java
/*package com.chad.handle.audio;

import com.chad.handle.TrackScheduler;
import com.chad.handle.audio.AudioProvider;
import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.player.AudioPlayerManager;

public class MusicHandler {
    public final AudioPlayerManager manager;
	public final AudioPlayer player;
	public final TrackScheduler scheduler;

	public MusicHandler(AudioPlayerManager manager) {
	    this.manager = manager;
		player = manager.createPlayer();
		scheduler = new TrackScheduler(player);
		player.addListener(scheduler);
	}

	public AudioProvider getAudioProvider() {
		return new AudioProvider(player);
	}
}*/

File: chad/src/main/java/dev/shog/chad/framework/handle/MusicHandler.kt
@file:JvmName("MusicHandler")

package dev.shog.chad.framework.handle

import dev.shog.chad.core.ChadVar.musicManagers
import dev.shog.chad.core.ChadVar.playerManager
import dev.shog.chad.framework.obj.GuildMusicManager
import sx.blah.discord.handle.obj.IGuild
import sx.blah.discord.handle.obj.IVoiceChannel

/**
 * Gets a guild's audio player
 */
fun getMusicManager(guild: IGuild, voiceChannel: IVoiceChannel): dev.shog.chad.framework.obj.GuildMusicManager {
    val guildId = guild.longID
    val channelId = voiceChannel.longID
    var musicManager: dev.shog.chad.framework.obj.GuildMusicManager? = musicManagers[guildId]

    if (musicManager == null) {
        musicManager = dev.shog.chad.framework.obj.GuildMusicManager(playerManager, guildId, channelId)

        musicManagers[guildId] = musicManager
    }

    musicManager.scheduler.channelId = voiceChannel.longID
    guild.audioManager.audioProvider = musicManager.audioProvider
    return musicManager
}

/**
 * This doesn't reset the music manager for the new channel, and just grabs the guild's instance.
 */
fun getMusicManager(guild: IGuild): dev.shog.chad.framework.obj.GuildMusicManager? {
    val guildId = guild.longID

    val musicManager = musicManagers[guildId] ?: return null
    guild.audioManager.audioProvider = musicManager.audioProvider
    return musicManager
}